Kitodo-DataManagement/src/main/java/org/kitodo/util/GeneratorSwitch.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * (c) Kitodo. Key to digital objects e. V. <contact@kitodo.org>
+ *
+ * This file is part of the Kitodo project.
+ *
+ * It is licensed under GNU General Public License version 3 or later.
+ *
+ * For the full copyright and license information, please read the
+ * GPL3-License.txt file that was distributed with this source code.
+ */
+
+package org.kitodo.util;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.kitodo.data.database.beans.Folder;
+import org.kitodo.data.database.beans.Project;
+import org.kitodo.data.database.beans.Task;
+
+/**
+ * An encapsulation to access the content creation properties of a task from
+ * JSF.
+ *
+ * <p>
+ * The content creation properties of a task are stored internally differently
+ * than they are displayed in the user interface. The user interface displays a
+ * switch for each folder, for which a generator function has been configured.
+ * Internally, however, the configuration of the activated generator functions
+ * is stored in a list of the task in which all folders are contained, in which
+ * the generator function is activated, that is, the switch is selected.
+ * Conversely, the task knows nothing about which non-selected folders with
+ * configured generator function there.
+ */
+public class GeneratorSwitch {
+    /**
+     * Folder represented by this generator switch.
+     */
+    private Folder folder;
+
+    /**
+     * Modifyable list containing enabled generators. This list is member of the
+     * {@link Task} and saves the generator state when the task is saved.
+     */
+    private List<Folder> contentFolders;
+
+    /**
+     * Returns a list of generator switches for all folders whose contents can
+     * be generated.
+     *
+     * @param projects
+     *            stream of projects this task is used in
+     * @param contentFolders
+     *            modifiable list of folders whose contents are to be generated
+     * @return list of GeneratorSwitch objects or empty list
+     */
+    @SuppressWarnings("serial")
+    public static List<GeneratorSwitch> getGeneratorSwitches(Stream<Project> projects, List<Folder> contentFolders) {
+
+        // Ignore all projects that do not have a source folder configured. It
```

Comment:
In general whole inserted code is without tests - add tests after code is placed in right place and redundancies are gone.